flink-examples/flink-examples-streaming/src/main/java/org/apache/flink/streaming/examples/sideoutput/SideOutputExample.java

```diff
@@ -148,4 +156,29 @@ public void processElement(
 
 		}
 	}
+
+	/**
+	 * This {@link WatermarkStrategy} assigns the current system time as the event-time timestamp.
+	 * In a real use case you should use proper timestamps and an appropriate {@link
+	 * WatermarkStrategy}.
+	 */
+	private static class IngestionTimeWatermarkStrategy<T> implements WatermarkStrategy<T> {
```

Comment:
This `WatermarkStrategy` seems to be used extensively in the tests. Couldn't we extract it somewhere and make it reusable?

Reply:
I had this in the commit message:
```
I removed calls to set IngestionTime and replaced them by an explicit
IngestionTimeWatermarkStrategy. I duplicated the same
IngestionTimeWatermarkStrategy in all the examples/tests because I
explicitly didn't want to add an IngestionTimeWatermarkStrategy in one
of the core packages so that it is not discoverable because I think we
shouldn't encourage users to use ingestion time.
```

what do you think?

Reply:
I was thinking more of a testing module, like the `MockSource`, but I think you are right to not expose it even there.